src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/AppointmentDtos.kt

```diff
@@ -20,16 +21,68 @@ data class CreateBookingAppointment(
   val locationId: Long,
   val comment: String? = null,
   val startTime: String,
-  val endTime: String
+  val endTime: String,
+  val repeat: Repeat? = null
 )
 
+@ApiModel(description = "The data required to create an appointment")
 data class CreateAppointmentSpecification(
-  val bookingId: Long? = null,
+  @ApiModelProperty(required = true, value = "The offender booking id")
+  val bookingId: Long,
+  @ApiModelProperty(required = true, value = "The location id of where the appointment will take place")
   val locationId: Long,
+  @ApiModelProperty(required = true, value = "Appointment type", example = "INST")
   val appointmentType: String,
+  @ApiModelProperty(required = false, value = "Additional information")
   val comment: String? = null,
+  @ApiModelProperty(
+    required = true,
+    value = "The date and time the appointment is scheduled for",
+    example = "2021-05-23T17:00:00"
+  )
   val startTime: LocalDateTime,
-  val endTime: LocalDateTime? = null
+  @ApiModelProperty(
+    required = false,
+    value = "The estimated date time the appointment will end",
+    example = "2021-05-23T17:00:00"
+  )
+  val endTime: LocalDateTime? = null,
+  @ApiModelProperty(required = false, value = "Describes how many times this appointment is to be repeated")
+  val repeat: Repeat? = null
+)
+
+data class CreatePrisonAppointment(
```

Comment:
Maybe this is for later refactoring, but this data structure (and I suspect others in this file) are not DTOs.. and this class is getting too big IMO... so we ought to put them in a class (prob in the same package as the service)

Reply:
Agreed